File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.util.stream.IntStream;

class ArmstrongNumbers {
  boolean isArmstrongNumber(int numberToCheck) {
    char[] chars = String.valueOf(numberToCheck).toCharArray();
    int power = chars.length;
    return IntStream
        .range(0, power)
        .map(i -> (int)Math.pow(Character.getNumericValue(chars[i]), power))
        .sum() == numberToCheck;
  }
}

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
  public static bool IsArmstrongNumber(int n)
  {
    var chars = n.ToString().AsEnumerable();
    var pow = chars.Count();
    return n == chars.Select(x => Math.Pow(double.Parse(x.ToString()), pow)).Sum();
  }
}